rd_ui/app/scripts/directives/plotly.js

```diff
@@ -288,6 +311,25 @@
           var element = element[0].children[0];
           Plotly.newPlot(element, scope.data, scope.layout, scope.plotlyOptions);
 
+          element.on('plotly_afterplot', function(d) {
+            if(scope.options.globalSeriesType === 'area' && (scope.options.series.stacking === 'normal' || scope.options.series.stacking === 'percent')){
+              d3.selectAll('.legendtoggle').each(function(d, i) {
+                // override plotly's default legend item on-click behaviour
```

Comment:
Thank you for the suggestion. We went back and took a look at it. It turns out we can use a custom listener through d3, and remove it when chart type changes.